Automattic/wp-calypso
# Plugins: Add me sites plugins endpoint request

#### Changes proposed in this Pull Request

Alternative to https://github.com/Automattic/wp-calypso/pull/61747 

It adds the `/me/sites/plugins` endpoint request every time we make a request for all plugins. 
This improves the performance of the /plugins/manage/* pages. Since before that we would be making a request for each individual site which makes this page not usable for sites that have more than a handful of sites. 

This new implementation uses `react-query` to store the request details instead of the current redux store.

#### Testing instructions

* Visit /plugins/manage notice that the screen loads as expected (hopefully more instantly as well ) 
* Notice that the values match the current ones in production. ( If they don't it is not really related to this PR, but since the source of the data is different ( cache site vs the site directly it is good to pay attention to this) 
* install, activate, enable autoupdate, update and remove plugins still works as expected. 
* visit /plugins/akismet notice that the screen loads as expected. ( hopefully more instantly ) 
* visit /activity-log/yourjetpacksite.com you should see plugins that need updating here. 
* visit /plans/my-plan/yourjetpacksite.com notice that the "Automattic plugin updates" tour still works as expected.
